diff --git a/node_modules/@react-navigation/core/lib/commonjs/getPathFromState.js b/node_modules/@react-navigation/core/lib/commonjs/getPathFromState.js
index dbed7e2..eca8f9b 100644
--- a/node_modules/@react-navigation/core/lib/commonjs/getPathFromState.js
+++ b/node_modules/@react-navigation/core/lib/commonjs/getPathFromState.js
@@ -45,9 +45,10 @@ const getActiveRoute = state => {
  *
  * @param state Navigation state to serialize.
  * @param options Extra options to fine-tune how to serialize the path.
+ * @param isTabScreen Return true if it's a TabNavigator screen
  * @returns Path representing the state, e.g. /foo/bar?count=42.
  */
-function getPathFromState(state, options) {
+function getPathFromState(state, options, isTabScreen) {
   if (state == null) {
     throw Error("Got 'undefined' for the navigation state. You must pass a valid state object.");
   }
@@ -68,6 +69,10 @@ function getPathFromState(state, options) {
     let focusedRoute = getActiveRoute(state);
     let currentOptions = configs;
 
+    if (isTabScreen && isTabScreen(route.name)) {
+      currentOptions = configs?.TabNavigator?.screens || configs
+    }
+
     // Keep all the route names that appeared during going deeper in config in case the pattern is resolved to undefined
     let nestedRouteNames = [];
     let hasNext = true;
diff --git a/node_modules/@react-navigation/core/lib/module/getPathFromState.js b/node_modules/@react-navigation/core/lib/module/getPathFromState.js
index 1cf84f2..b9ade7b 100644
--- a/node_modules/@react-navigation/core/lib/module/getPathFromState.js
+++ b/node_modules/@react-navigation/core/lib/module/getPathFromState.js
@@ -36,9 +36,10 @@ const getActiveRoute = state => {
  *
  * @param state Navigation state to serialize.
  * @param options Extra options to fine-tune how to serialize the path.
+ * @param isTabScreen Return true if it's a TabNavigator screen
  * @returns Path representing the state, e.g. /foo/bar?count=42.
  */
-export default function getPathFromState(state, options) {
+export default function getPathFromState(state, options, isTabScreen) {
   if (state == null) {
     throw Error("Got 'undefined' for the navigation state. You must pass a valid state object.");
   }
@@ -59,6 +60,10 @@ export default function getPathFromState(state, options) {
     let focusedRoute = getActiveRoute(state);
     let currentOptions = configs;
 
+    if (isTabScreen && isTabScreen(route.name)) {
+      currentOptions = configs?.TabNavigator?.screens || configs
+    }
+
     // Keep all the route names that appeared during going deeper in config in case the pattern is resolved to undefined
     let nestedRouteNames = [];
     let hasNext = true;
diff --git a/node_modules/@react-navigation/core/lib/typescript/src/getPathFromState.d.ts b/node_modules/@react-navigation/core/lib/typescript/src/getPathFromState.d.ts
index 6b78f6c..d13d622 100644
--- a/node_modules/@react-navigation/core/lib/typescript/src/getPathFromState.d.ts
+++ b/node_modules/@react-navigation/core/lib/typescript/src/getPathFromState.d.ts
@@ -32,8 +32,9 @@ type State = NavigationState | Omit<PartialState<NavigationState>, 'stale'>;
  *
  * @param state Navigation state to serialize.
  * @param options Extra options to fine-tune how to serialize the path.
+ * @param isTabScreen Return true if it's a TabNavigator screen
  * @returns Path representing the state, e.g. /foo/bar?count=42.
  */
-export default function getPathFromState<ParamList extends {}>(state: State, options?: Options<ParamList>): string;
+export default function getPathFromState<ParamList extends {}>(state: State, options?: Options<ParamList>, isTabScreen?: (name: string) => boolean): string;
 export {};
 //# sourceMappingURL=getPathFromState.d.ts.map
\ No newline at end of file
diff --git a/node_modules/@react-navigation/core/src/getPathFromState.tsx b/node_modules/@react-navigation/core/src/getPathFromState.tsx
index 26a6213..02b2dd7 100644
--- a/node_modules/@react-navigation/core/src/getPathFromState.tsx
+++ b/node_modules/@react-navigation/core/src/getPathFromState.tsx
@@ -64,11 +64,13 @@ const getActiveRoute = (state: State): { name: string; params?: object } => {
  *
  * @param state Navigation state to serialize.
  * @param options Extra options to fine-tune how to serialize the path.
+ * @param isTabScreen Return true if it's a TabNavigator screen
  * @returns Path representing the state, e.g. /foo/bar?count=42.
  */
 export default function getPathFromState<ParamList extends {}>(
   state: State,
-  options?: Options<ParamList>
+  options?: Options<ParamList>,
+  isTabScreen?: (name: string) => boolean
 ): string {
   if (state == null) {
     throw Error(
@@ -102,6 +104,10 @@ export default function getPathFromState<ParamList extends {}>(
     let focusedRoute = getActiveRoute(state);
     let currentOptions = configs;
 
+    if (isTabScreen && isTabScreen(route.name as any)) {
+      currentOptions = configs?.TabNavigator?.screens as any || configs
+    }
+
     // Keep all the route names that appeared during going deeper in config in case the pattern is resolved to undefined
     let nestedRouteNames = [];
 
