diff --git a/node_modules/@react-navigation/native/lib/commonjs/createMemoryHistory.js b/node_modules/@react-navigation/native/lib/commonjs/createMemoryHistory.js
index dfdec7f..24d63ff 100644
--- a/node_modules/@react-navigation/native/lib/commonjs/createMemoryHistory.js
+++ b/node_modules/@react-navigation/native/lib/commonjs/createMemoryHistory.js
@@ -34,7 +34,7 @@ function createMemoryHistory() {
       const id = (_window$history$state = window.history.state) === null || _window$history$state === void 0 ? void 0 : _window$history$state.id;
 
       if (id) {
-        const index = items.findIndex(item => item.id === id);
+        const index = items.findIndex(item => item?.id === id);
         return index > -1 ? index : 0;
       }
 
@@ -97,7 +97,7 @@ function createMemoryHistory() {
       interrupt();
       const id = (_window$history$state2 = (_window$history$state3 = window.history.state) === null || _window$history$state3 === void 0 ? void 0 : _window$history$state3.id) !== null && _window$history$state2 !== void 0 ? _window$history$state2 : (0, _nonSecure.nanoid)();
 
-      if (!items.length || items.findIndex(item => item.id === id) < 0) {
+      if (!items.length || items.findIndex(item => item?.id === id) < 0) {
         // There are two scenarios for creating an array with only one history record:
         // - When loaded id not found in the items array, this function by default will replace
         //   the first item. We need to keep only the new updated object, otherwise it will break
@@ -203,7 +203,7 @@ function createMemoryHistory() {
           var _window$history$state4;
 
           const id = (_window$history$state4 = window.history.state) === null || _window$history$state4 === void 0 ? void 0 : _window$history$state4.id;
-          const currentIndex = items.findIndex(item => item.id === id); // Fix createMemoryHistory.index variable's value
+          const currentIndex = items.findIndex(item => item?.id === id); // Fix createMemoryHistory.index variable's value
           // as it may go out of sync when navigating in the browser.
 
           index = Math.max(currentIndex, 0);
diff --git a/node_modules/@react-navigation/native/lib/module/createMemoryHistory.js b/node_modules/@react-navigation/native/lib/module/createMemoryHistory.js
index 2b024a6..5f86067 100644
--- a/node_modules/@react-navigation/native/lib/module/createMemoryHistory.js
+++ b/node_modules/@react-navigation/native/lib/module/createMemoryHistory.js
@@ -26,7 +26,7 @@ export default function createMemoryHistory() {
       const id = (_window$history$state = window.history.state) === null || _window$history$state === void 0 ? void 0 : _window$history$state.id;
 
       if (id) {
-        const index = items.findIndex(item => item.id === id);
+        const index = items.findIndex(item => item?.id === id);
         return index > -1 ? index : 0;
       }
 
@@ -89,7 +89,7 @@ export default function createMemoryHistory() {
       interrupt();
       const id = (_window$history$state2 = (_window$history$state3 = window.history.state) === null || _window$history$state3 === void 0 ? void 0 : _window$history$state3.id) !== null && _window$history$state2 !== void 0 ? _window$history$state2 : nanoid();
 
-      if (!items.length || items.findIndex(item => item.id === id) < 0) {
+      if (!items.length || items.findIndex(item => item?.id === id) < 0) {
         // There are two scenarios for creating an array with only one history record:
         // - When loaded id not found in the items array, this function by default will replace
         //   the first item. We need to keep only the new updated object, otherwise it will break
@@ -195,7 +195,7 @@ export default function createMemoryHistory() {
           var _window$history$state4;
 
           const id = (_window$history$state4 = window.history.state) === null || _window$history$state4 === void 0 ? void 0 : _window$history$state4.id;
-          const currentIndex = items.findIndex(item => item.id === id); // Fix createMemoryHistory.index variable's value
+          const currentIndex = items.findIndex(item => item?.id === id); // Fix createMemoryHistory.index variable's value
           // as it may go out of sync when navigating in the browser.
 
           index = Math.max(currentIndex, 0);
diff --git a/node_modules/@react-navigation/native/src/createMemoryHistory.tsx b/node_modules/@react-navigation/native/src/createMemoryHistory.tsx
index 515b0da..2a36336 100644
--- a/node_modules/@react-navigation/native/src/createMemoryHistory.tsx
+++ b/node_modules/@react-navigation/native/src/createMemoryHistory.tsx
@@ -35,7 +35,7 @@ export default function createMemoryHistory() {
       const id = window.history.state?.id;
 
       if (id) {
-        const index = items.findIndex((item) => item.id === id);
+        const index = items.findIndex((item) => item?.id === id);
 
         return index > -1 ? index : 0;
       }
@@ -84,7 +84,7 @@ export default function createMemoryHistory() {
 
       const id = window.history.state?.id ?? nanoid();
 
-      if (!items.length || items.findIndex((item) => item.id === id) < 0) {
+      if (!items.length || items.findIndex((item) => item?.id === id) < 0) {
         // There are two scenarios for creating an array with only one history record:
         // - When loaded id not found in the items array, this function by default will replace
         //   the first item. We need to keep only the new updated object, otherwise it will break
@@ -176,7 +176,7 @@ export default function createMemoryHistory() {
 
         const onPopState = () => {
           const id = window.history.state?.id;
-          const currentIndex = items.findIndex((item) => item.id === id);
+          const currentIndex = items.findIndex((item) => item?.id === id);
 
           // Fix createMemoryHistory.index variable's value
           // as it may go out of sync when navigating in the browser.
