diff --git a/node_modules/@wdio/cli/build/launcher.js b/node_modules/@wdio/cli/build/launcher.js
index acb6d97..dadbc32 100644
--- a/node_modules/@wdio/cli/build/launcher.js
+++ b/node_modules/@wdio/cli/build/launcher.js
@@ -29,6 +29,17 @@ class Launcher {
          */
         if (_args.autoCompileOpts) {
             this.configParser.merge({ autoCompileOpts: _args.autoCompileOpts });
+        } else {
+            // added else bloc until https://github.com/webdriverio/webdriverio/issues/9151 is fixed
+            this.configParser.merge({
+                autoCompileOpts: {
+                    autoCompile: true,
+                    tsNodeOpts: {
+                        transpileOnly: true,
+                        project: process.cwd() + '/e2e/tsconfig.json'
+                    },
+                },
+            });
         }
         this.configParser.autoCompile();
         this.configParser.addConfigFile(_configFilePath);
