diff --git a/node_modules/algoliasearch/index.js b/node_modules/algoliasearch/index.js
index 14ca1ec..ba72c91 100644
--- a/node_modules/algoliasearch/index.js
+++ b/node_modules/algoliasearch/index.js
@@ -1,5 +1,5 @@
 /* eslint-disable functional/immutable-data, import/no-commonjs */
-const algoliasearch = require('./dist/algoliasearch.cjs.js');
+const algoliasearch = require('./dist/algoliasearch-lite.esm.browser.js');
 
 /**
  * The Common JS build is the default entry point for the Node environment. Keep in
