diff --git a/node_modules/eslint-plugin-import/lib/rules/no-extraneous-dependencies.js b/node_modules/eslint-plugin-import/lib/rules/no-extraneous-dependencies.js
index 6a14031..65dce65 100644
--- a/node_modules/eslint-plugin-import/lib/rules/no-extraneous-dependencies.js
+++ b/node_modules/eslint-plugin-import/lib/rules/no-extraneous-dependencies.js
@@ -179,6 +179,7 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
 
   if (
   declarationStatus.isInDeps ||
+  depsOptions.allowFolder  ||
   depsOptions.allowDevDeps && declarationStatus.isInDevDeps ||
   depsOptions.allowPeerDeps && declarationStatus.isInPeerDeps ||
   depsOptions.allowOptDeps && declarationStatus.isInOptDeps ||
@@ -195,6 +196,7 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
 
     if (
     declarationStatus.isInDeps ||
+    depsOptions.allowFolder  ||
     depsOptions.allowDevDeps && declarationStatus.isInDevDeps ||
     depsOptions.allowPeerDeps && declarationStatus.isInPeerDeps ||
     depsOptions.allowOptDeps && declarationStatus.isInOptDeps ||
@@ -229,6 +231,10 @@ function testConfig(config, filename) {
 
 }
 
+function notInSrc(filename) {
+  return (filename.includes("tests") || filename.includes("src"))
+}
+
 module.exports = {
   meta: {
     type: 'problem',
@@ -260,7 +266,8 @@ module.exports = {
         allowDevDeps: testConfig(options.devDependencies, filename) !== false,
         allowOptDeps: testConfig(options.optionalDependencies, filename) !== false,
         allowPeerDeps: testConfig(options.peerDependencies, filename) !== false,
-        allowBundledDeps: testConfig(options.bundledDependencies, filename) !== false };
+        allowBundledDeps: testConfig(options.bundledDependencies, filename) !== false,
+        allowFolder: notInSrc(filename) };
 
 
       return (0, _moduleVisitor2['default'])(function (source, node) {
