diff --git a/node_modules/react-native-web/dist/cjs/exports/Image/index.js b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
index 945174e..559ad3e 100644
--- a/node_modules/react-native-web/dist/cjs/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
@@ -188,6 +188,7 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
       pointerEvents = props.pointerEvents,
       source = props.source,
       style = props.style,
+      transform = props.transform,
       rest = _objectWithoutPropertiesLoose(props, ["accessibilityLabel", "blurRadius", "defaultSource", "draggable", "onError", "onLayout", "onLoad", "onLoadEnd", "onLoadStart", "pointerEvents", "source", "style"]);
 
   if (process.env.NODE_ENV !== 'production') {
@@ -326,7 +327,8 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
   }), /*#__PURE__*/React.createElement(_View.default, {
     style: [styles.image, resizeModeStyles[resizeMode], {
       backgroundImage: backgroundImage,
-      filter: filter
+      filter: filter,
+      transform: transform
     }, backgroundSize != null && {
       backgroundSize: backgroundSize
     }],
diff --git a/node_modules/react-native-web/dist/exports/Image/index.js b/node_modules/react-native-web/dist/exports/Image/index.js
index 5c25ebf..b9ab20d 100644
--- a/node_modules/react-native-web/dist/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/exports/Image/index.js
@@ -175,6 +175,7 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
       pointerEvents = props.pointerEvents,
       source = props.source,
       style = props.style,
+      transform = props.transform,
       rest = _objectWithoutPropertiesLoose(props, ["accessibilityLabel", "blurRadius", "defaultSource", "draggable", "onError", "onLayout", "onLoad", "onLoadEnd", "onLoadStart", "pointerEvents", "source", "style"]);
 
   if (process.env.NODE_ENV !== 'production') {
@@ -312,7 +313,8 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
   }), /*#__PURE__*/React.createElement(View, {
     style: [styles.image, resizeModeStyles[resizeMode], {
       backgroundImage: backgroundImage,
-      filter: filter
+      filter: filter,
+      transform: transform
     }, backgroundSize != null && {
       backgroundSize: backgroundSize
     }],
diff --git a/node_modules/react-native-web/src/exports/Image/index.js b/node_modules/react-native-web/src/exports/Image/index.js
index 7cbde9f..5cc166c 100644
--- a/node_modules/react-native-web/src/exports/Image/index.js
+++ b/node_modules/react-native-web/src/exports/Image/index.js
@@ -157,6 +157,7 @@ const Image: React.AbstractComponent<ImageProps, React.ElementRef<typeof View>>
       pointerEvents,
       source,
       style,
+      transform,
       ...rest
     } = props;
 
@@ -290,7 +291,7 @@ const Image: React.AbstractComponent<ImageProps, React.ElementRef<typeof View>>
           style={[
             styles.image,
             resizeModeStyles[resizeMode],
-            { backgroundImage, filter },
+            { backgroundImage, filter, transform },
             backgroundSize != null && { backgroundSize }
           ]}
           suppressHydrationWarning={true}
