diff --git a/node_modules/react-native-web/dist/cjs/exports/Image/index.js b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
index 945174e..eaf3f4b 100644
--- a/node_modules/react-native-web/dist/cjs/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/Image/index.js
@@ -188,6 +188,7 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
       pointerEvents = props.pointerEvents,
       source = props.source,
       style = props.style,
+      transform = props.transform,
       rest = _objectWithoutPropertiesLoose(props, ["accessibilityLabel", "blurRadius", "defaultSource", "draggable", "onError", "onLayout", "onLoad", "onLoadEnd", "onLoadStart", "pointerEvents", "source", "style"]);
 
   if (process.env.NODE_ENV !== 'production') {
@@ -326,7 +327,8 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
   }), /*#__PURE__*/React.createElement(_View.default, {
     style: [styles.image, resizeModeStyles[resizeMode], {
       backgroundImage: backgroundImage,
-      filter: filter
+      filter: filter,
+      transform: transform
     }, backgroundSize != null && {
       backgroundSize: backgroundSize
     }],
diff --git a/node_modules/react-native-web/dist/cjs/exports/Text/index.js b/node_modules/react-native-web/dist/cjs/exports/Text/index.js
index 36ad30c..e91e233 100644
--- a/node_modules/react-native-web/dist/cjs/exports/Text/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/Text/index.js
@@ -110,7 +110,7 @@ var Text = /*#__PURE__*/React.forwardRef(function (props, forwardedRef) {
   supportedProps.dir = dir; // 'auto' by default allows browsers to infer writing direction (root elements only)
 
   if (!hasTextAncestor) {
-    supportedProps.dir = dir != null ? dir : 'auto';
+    supportedProps.dir = dir != null ? dir : 'ltr';
   }
 
   if (onClick || onPress) {
diff --git a/node_modules/react-native-web/dist/cjs/exports/TextInput/index.js b/node_modules/react-native-web/dist/cjs/exports/TextInput/index.js
index e93533a..227312b 100644
--- a/node_modules/react-native-web/dist/cjs/exports/TextInput/index.js
+++ b/node_modules/react-native-web/dist/cjs/exports/TextInput/index.js
@@ -365,7 +365,7 @@ var TextInput = /*#__PURE__*/React.forwardRef(function (props, forwardedRef) {
   supportedProps.autoCorrect = autoCorrect ? 'on' : 'off';
   supportedProps.classList = classList; // 'auto' by default allows browsers to infer writing direction
 
-  supportedProps.dir = dir !== undefined ? dir : 'auto';
+  supportedProps.dir = dir !== undefined ? dir : 'ltr';
   supportedProps.enterKeyHint = returnKeyType;
   supportedProps.onBlur = handleBlur;
   supportedProps.onChange = handleChange;
diff --git a/node_modules/react-native-web/dist/exports/Image/index.js b/node_modules/react-native-web/dist/exports/Image/index.js
index 5c25ebf..f360c47 100644
--- a/node_modules/react-native-web/dist/exports/Image/index.js
+++ b/node_modules/react-native-web/dist/exports/Image/index.js
@@ -175,6 +175,7 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
       pointerEvents = props.pointerEvents,
       source = props.source,
       style = props.style,
+      transform = props.transform,
       rest = _objectWithoutPropertiesLoose(props, ["accessibilityLabel", "blurRadius", "defaultSource", "draggable", "onError", "onLayout", "onLoad", "onLoadEnd", "onLoadStart", "pointerEvents", "source", "style"]);
 
   if (process.env.NODE_ENV !== 'production') {
@@ -312,7 +313,8 @@ var Image = /*#__PURE__*/React.forwardRef(function (props, ref) {
   }), /*#__PURE__*/React.createElement(View, {
     style: [styles.image, resizeModeStyles[resizeMode], {
       backgroundImage: backgroundImage,
-      filter: filter
+      filter: filter,
+      transform: transform
     }, backgroundSize != null && {
       backgroundSize: backgroundSize
     }],
diff --git a/node_modules/react-native-web/dist/exports/Text/index.js b/node_modules/react-native-web/dist/exports/Text/index.js
index 034bbd1..62490b6 100644
--- a/node_modules/react-native-web/dist/exports/Text/index.js
+++ b/node_modules/react-native-web/dist/exports/Text/index.js
@@ -98,7 +98,7 @@ var Text = /*#__PURE__*/React.forwardRef(function (props, forwardedRef) {
   supportedProps.dir = dir; // 'auto' by default allows browsers to infer writing direction (root elements only)
 
   if (!hasTextAncestor) {
-    supportedProps.dir = dir != null ? dir : 'auto';
+    supportedProps.dir = dir != null ? dir : 'ltr';
   }
 
   if (onClick || onPress) {
diff --git a/node_modules/react-native-web/dist/exports/TextInput/index.js b/node_modules/react-native-web/dist/exports/TextInput/index.js
index d55b552..1d968da 100644
--- a/node_modules/react-native-web/dist/exports/TextInput/index.js
+++ b/node_modules/react-native-web/dist/exports/TextInput/index.js
@@ -350,7 +350,7 @@ var TextInput = /*#__PURE__*/React.forwardRef(function (props, forwardedRef) {
   supportedProps.autoCorrect = autoCorrect ? 'on' : 'off';
   supportedProps.classList = classList; // 'auto' by default allows browsers to infer writing direction
 
-  supportedProps.dir = dir !== undefined ? dir : 'auto';
+  supportedProps.dir = dir !== undefined ? dir : 'ltr';
   supportedProps.enterKeyHint = returnKeyType;
   supportedProps.onBlur = handleBlur;
   supportedProps.onChange = handleChange;
diff --git a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
index 05a53ed..7d96447 100644
--- a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
+++ b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
@@ -1127,6 +1127,14 @@ var VirtualizedList = /*#__PURE__*/function (_React$PureComponent) {
     });
 
     this._hasMore = this.state.last < this.props.getItemCount(this.props.data) - 1;
+
+    const headerIdx = cells.findIndex((cell) => cell.key === "$header")
+    const footerIdx = cells.findIndex((cell) => cell.key === "$footer")
+    const cellsWithoutHeaderAndFooter = cells.filter((_,idx) => !([headerIdx, footerIdx].includes(idx)))
+    const isListWithNonNullItems = cellsWithoutHeaderAndFooter && this.props.listAs
+    const cellsAsListItem = isListWithNonNullItems ? React.createElement(this.props.listAs, {}, cellsWithoutHeaderAndFooter) : null
+    const cellsToDisplay = isListWithNonNullItems ? [cells[headerIdx], cellsAsListItem, cells[footerIdx]] : cells
+
     var innerRet = /*#__PURE__*/React.createElement(VirtualizedListContextProvider, {
       value: {
         cellKey: null,
@@ -1140,7 +1148,7 @@ var VirtualizedList = /*#__PURE__*/function (_React$PureComponent) {
       }
     }, /*#__PURE__*/React.cloneElement((this.props.renderScrollComponent || this._defaultRenderScrollComponent)(scrollProps), {
       ref: this._captureScrollRef
-    }, cells));
+    }, cellsToDisplay));
     var ret = innerRet;
 
     if (this.props.debug) {
@@ -1581,9 +1589,10 @@ var CellRenderer = /*#__PURE__*/function (_React$Component) {
       style: cellStyle,
       onLayout: onLayout
     }), element, itemSeparator);
+
     return /*#__PURE__*/React.createElement(VirtualizedListCellContextProvider, {
       cellKey: this.props.cellKey
-    }, result);
+    }, this.props.parentProps.itemAs ? React.createElement(this.props.parentProps.itemAs, {}, result) : result);
   };
 
   return CellRenderer;
diff --git a/node_modules/react-native-web/src/exports/Image/index.js b/node_modules/react-native-web/src/exports/Image/index.js
index 7cbde9f..5cc166c 100644
--- a/node_modules/react-native-web/src/exports/Image/index.js
+++ b/node_modules/react-native-web/src/exports/Image/index.js
@@ -157,6 +157,7 @@ const Image: React.AbstractComponent<ImageProps, React.ElementRef<typeof View>>
       pointerEvents,
       source,
       style,
+      transform,
       ...rest
     } = props;
 
@@ -290,7 +291,7 @@ const Image: React.AbstractComponent<ImageProps, React.ElementRef<typeof View>>
           style={[
             styles.image,
             resizeModeStyles[resizeMode],
-            { backgroundImage, filter },
+            { backgroundImage, filter, transform },
             backgroundSize != null && { backgroundSize }
           ]}
           suppressHydrationWarning={true}
diff --git a/node_modules/react-native-web/src/exports/Text/index.js b/node_modules/react-native-web/src/exports/Text/index.js
index a2a74ce..a70df07 100644
--- a/node_modules/react-native-web/src/exports/Text/index.js
+++ b/node_modules/react-native-web/src/exports/Text/index.js
@@ -121,7 +121,7 @@ const Text: React.AbstractComponent<TextProps, HTMLElement & PlatformMethods> =
     supportedProps.dir = dir;
     // 'auto' by default allows browsers to infer writing direction (root elements only)
     if (!hasTextAncestor) {
-      supportedProps.dir = dir != null ? dir : 'auto';
+      supportedProps.dir = dir != null ? dir : 'ltr';
     }
 
     if (onClick || onPress) {
diff --git a/node_modules/react-native-web/src/exports/TextInput/index.js b/node_modules/react-native-web/src/exports/TextInput/index.js
index a37dd8c..61df3b3 100644
--- a/node_modules/react-native-web/src/exports/TextInput/index.js
+++ b/node_modules/react-native-web/src/exports/TextInput/index.js
@@ -339,7 +339,7 @@ const TextInput: React.AbstractComponent<
   supportedProps.autoCorrect = autoCorrect ? 'on' : 'off';
   supportedProps.classList = classList;
   // 'auto' by default allows browsers to infer writing direction
-  supportedProps.dir = dir !== undefined ? dir : 'auto';
+  supportedProps.dir = dir !== undefined ? dir : 'ltr';
   supportedProps.enterKeyHint = returnKeyType;
   supportedProps.onBlur = handleBlur;
   supportedProps.onChange = handleChange;
