diff --git a/node_modules/react-native-performance-stats/index.ts b/node_modules/react-native-performance-stats/index.ts
new file mode 100644
index 0000000..a2ebcde
--- /dev/null
+++ b/node_modules/react-native-performance-stats/index.ts
@@ -0,0 +1 @@
+export { default } from 'react-native-performance-stats/src/index.tsx'
diff --git a/node_modules/react-native-performance-stats/src/NativePerformanceStats.js b/node_modules/react-native-performance-stats/src/NativePerformanceStats.js
deleted file mode 100644
index 01464e9..0000000
--- a/node_modules/react-native-performance-stats/src/NativePerformanceStats.js
+++ /dev/null
@@ -1,11 +0,0 @@
-// @flow
-import type { TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
-import { TurboModuleRegistry } from 'react-native';
-
-export interface Spec extends TurboModule {
-  start(withCPU?: boolean): void;
-  stop(): void;
-}
-export default (TurboModuleRegistry.get<Spec>(
-  'PerformanceStats'
-): ?Spec);
diff --git a/node_modules/react-native-performance-stats/src/NativePerformanceStats.ts b/node_modules/react-native-performance-stats/src/NativePerformanceStats.ts
new file mode 100644
index 0000000..d0b1b62
--- /dev/null
+++ b/node_modules/react-native-performance-stats/src/NativePerformanceStats.ts
@@ -0,0 +1,8 @@
+import type { TurboModule } from 'react-native';
+import { TurboModuleRegistry } from 'react-native';
+
+export interface Spec extends TurboModule {
+  start(withCPU?: boolean): void;
+  stop(): void;
+}
+export default TurboModuleRegistry.getEnforcing<Spec>('PerformanceStats');
\ No newline at end of file
diff --git a/node_modules/react-native-performance-stats/src/index.js b/node_modules/react-native-performance-stats/src/index.tsx
similarity index 99%
rename from node_modules/react-native-performance-stats/src/index.js
rename to node_modules/react-native-performance-stats/src/index.tsx
index b31e1a2..edc8e11 100644
